### PR TITLE
Add new performance test Merchant / Search Orders

### LINKED ISSUE #30637: k6 tests: Merchant - Search Orders
Create k6 test for the flow: Merchant - Search Orders

### PR DESCRIPTION
**Pull request by veljkho**
This is in progress, please review @tammullen but I expect that this require more work.

### Changes proposed in this Pull Request:

Closes #30637

### How to test the changes in this Pull Request:

- docker run --network="host" -v [WC ROOT PATH]/tests:/tests -it loadimpact/k6 run /tests/performance/tests/simple-all-merchant-requests.js

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: plugins/woocommerce/tests/performance/requests/merchant/orders-search.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	group("Orders Search", function () {
+		var requestHeaders = Object.assign({},
+			htmlRequestHeader,
+			commonRequestHeaders,
+			commonGetRequestHeaders,
+			commonNonStandardHeaders
+		);
+
+		response = http.get(
+			`${base_url}/wp-admin/edit.php?s&` +
```
Comment (tammullen) [inline plugins/woocommerce/tests/performance/requests/merchant/orders-search.js]: Request is missing search term
```suggestion
			`${base_url}/wp-admin/edit.php?s=${product_search_term}&` +
```

File: plugins/woocommerce/tests/performance/requests/merchant/orders-search.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+			`post_status=all&post_type=shop_order&action=-1&m=0&_customer_user&` +
+			`paged=1&action2=-1`,
+			{
+				headers: requestHeaders,
+			}
+		);
+		postTypeOrderSearchTrend.add(response.timings.duration);
+		check(response, {
+			"is status 200": (r) => r.status === 200,
+			"body contains: 'Orders' header": (response) =>
```
Comment (tammullen) [inline plugins/woocommerce/tests/performance/requests/merchant/orders-search.js]: Check might be better to be specific to search results

```suggestion
			"body contains: 'Search results' subtitle": (response) =>
```

File: plugins/woocommerce/tests/performance/requests/merchant/orders-search.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+				response.body.includes("Orders</h1>"),
```
Comment (tammullen) [inline plugins/woocommerce/tests/performance/requests/merchant/orders-search.js]:
```suggestion
				response.body.includes("Search results for:"),
```

**Review by tammullen — CHANGES_REQUESTED** [review]
Thanks for this! Sorry about the delay. Just needs some changes to include search term in request and update the checks.

**Comment by veljkho** [comment]
Thanks @tammullen for reviewing, I did necessary changes! Please review it again. Thanks!

**Comment by veljkho** [comment]
Hi @tammullen 
Can we merge this pull request?

File: plugins/woocommerce/tests/performance/requests/merchant/orders-search.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
+import { sleep, check, group } from "k6";
+import http from "k6/http";
+import { Trend } from "k6/metrics";
+import { randomIntBetween } from "https://jslib.k6.io/k6-utils/1.1.0/index.js";
+import { base_url, think_time_min, think_time_max } from "../../config.js";
```
Comment (tammullen) [inline plugins/woocommerce/tests/performance/requests/merchant/orders-search.js]: `product_search_term` needs to be included in this import

**Review by tammullen — CHANGES_REQUESTED** [review]
Sorry for the delay! Changes look good! Just needs one more fix to run ok

File: plugins/woocommerce/tests/performance/requests/merchant/orders-search.js (hunk 4 above)
Comment (veljkho) [reply plugins/woocommerce/tests/performance/requests/merchant/orders-search.js]: Thanks, that was missing. Added it now, please just confirm so I could merge it after.

**Review by tammullen — APPROVED** [review]
LGTM! Thanks for adding this test!

```
     █ Orders Search

       ✓ is status 200
       ✓ body contains: 'Search results' subtitle
       wc_get_post_type_order_search..............: avg=253.637   min=253.637  med=253.637  max=253.637  p(90)=253.637  p(95)=253.637  
```

**Pull request merged**